internal/services/v1/experimental.go

```diff
@@ -95,13 +82,18 @@ func NewExperimentalServer(dispatch dispatch.Dispatcher, permServerConfig Permis
 				streamtimeout.MustStreamServerInterceptor(config.StreamReadTimeout),
 			),
 		},
-		defaultBatchSize:        uint64(config.DefaultExportBatchSize),
-		maxBatchSize:            uint64(config.MaxExportBatchSize),
-		dispatch:                dispatch,
-		maximumAPIDepth:         permServerConfig.MaximumAPIDepth,
-		maxCaveatContextSize:    permServerConfig.MaxCaveatContextSize,
-		bulkCheckMaxConcurrency: config.BulkCheckMaxConcurrency,
+		defaultBatchSize: uint64(config.DefaultExportBatchSize),
+		maxBatchSize:     uint64(config.MaxExportBatchSize),
 	}
+	es.bulkChecker = &bulkChecker{
```

Comment:
assign this in the struct creation above

Reply:
I would have done that if I didn't need to pass the `rewriteError()` method.

Reply:
we should be rewriting errors at the call sites; so I'd remove that and just call rewriteError if/when an error occurs at the place where the shared method is invoked

Reply:
I tried that, but we have a [short-circuit in the bulk check implementation](https://github.com/authzed/spicedb/pull/1792/files#diff-4610fb4679843c5126b0e6b9a3b71f0649411314b9e122550f2506ec7cad1f9cR107-R112) which requires inspecting the result of `rewriteError()`; that's the only reason we are passing it in.

Reply:
Looking at the code, I guess we could just call `shared.RewriteError()` directly in the shared `checkBulkPermissions()` instead of passing in `rewriteError()`. Let me try that.